src/Workspaces/Remote/ServiceHub/CodeLensOOP/IRemoteCodeLensReferencesFromPrimaryWorkspaceService.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Remote.CodeLensOOP
+{
+    /// <summary>
+    /// new code lens API that runs on codelen's OOP assumes it gets result from roslyn OOP's primary workspace.
+    /// 
+    /// for now, we keep both new and old APIs, but before v.Next we will decide wether we will go this way or move back to old API
+    /// but add new ability in new code lens API where it can point to any OOP as code lens provider OOP host rather than forcing
+    /// people to use codelens OOP which might not have all information.
+    /// </summary>
+    internal interface IRemoteCodeLensReferencesFromPrimaryWorkspaceService
```

Comment:
so... quite possibly a dumb question... but why not just have a normal remote-service in OOP (like the reset of our remote services) that uses all the standard Roslyn/OOP concepts, and just asynchronously computes the results.

Then in VS itself, you can take those results and translate tehm to whatever the CodeLens system thinks it needs.

--

I ask because it's *super* weird to see these APIs that take guids/strings/etc. instead of things like Project/DocumentIds and whatnot.  It feels like we're leaking over the codelens ickyness to OOP, instead of keepign OOP clean and having the translation only occur in VS.  

Thoughts?

Reply:
that was how existing system worked. now we have code lens OOP that's in the middle which doesn't have any Roslyn OOP stuff in it. we know it is weird. but it is the way v1 works. we have request for v2 where Roslyn OOP can act as code lens OOP server rather than having code lens OOP as middle man. but for now, all call must go through code lens OOP which doesn't have roslyn OOP concepts. it uses servicehub API directly.